flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/TransformationScanProvider.java

```diff
@@ -41,11 +42,13 @@
 public interface TransformationScanProvider extends ScanTableSource.ScanRuntimeProvider {
 
     /** Helper method for creating a static provider. The boundedness is derived automatically. */
-    static TransformationScanProvider of(Transformation<RowData> transformation) {
+    static TransformationScanProvider of(
```

Comment:
I would rather suggest to remove the `of` method. Assigning a name only to the leaf transformation is never enough and can lead to side effects. This class is internal anyway.